src/graph/Interfaces.hpp

```diff
@@ -48,9 +48,15 @@ struct IPointsNode : public IPointCloudDescription
 	using Ptr = std::shared_ptr<IPointsNode>;
 
 	virtual VArray::ConstPtr getFieldData(rgl_field_t field, cudaStream_t stream) const = 0;
+	virtual std::size_t getFieldDataElemSize(rgl_field_t field, cudaStream_t stream) const = 0;
+	virtual std::size_t getFieldDataElemCount(rgl_field_t field, cudaStream_t stream) const = 0;
```

Comment:
Also, consider providing a default implementation to avoid boilerplate in other functions. This boilerplate is actually worse than having an if-else in one function. If you conclude that it cannot be easily solved, I'd stay with the if-else and fix this in the future together with a bigger refactor.